src/utils.ts

```diff
@@ -13,14 +13,11 @@ export const getTagForDeployment = (environment: string): string => {
   throw new Error('This action only supports push event on main branch or pull request events');
 };
 
-export const getLastDeployedRef = (environment: string, tagName: string): string => {
+export const getLastDeployedRef = (tagName: string): string => {
   debug(`Looking for last deployed ref - "${tagName}"`);
   const tagSha = getTagSha(tagName);
-  if (tagSha) {
-    return tagSha;
-  }
-  debug(`Tag was not found, deploy based on environment - "${environment}" `);
-  return getTagSha(environment);
+
+  return tagSha || 'refs/heads/main...';
```

Comment:
I think this works, but `...` is special syntax for diff so should we really call it a "ref"?

Maybe it makes more sense to return `undefined` and fall back on `refs/head/main...` in the caller, or perhaps to return `'${git merge-base main HEAD)'`, or to stop exporting `getLastDeployedRef` and make `getChangedPackages` call it (and send in the expected tag instead of a ref). Or turn this function into something like `buildGitDiffCommand`...